src/main/java/io/kestra/plugin/microsoft365/outlook/List.java

```diff
@@ -0,0 +1,263 @@
+                    auth:
+                      tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                      clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                      clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                      userPrincipalName: "user@example.com"
+                    folderId: "inbox"
+                    filter: "isRead eq false and receivedDateTime ge {{ now() | dateAdd(-7, 'DAYS') | date('yyyy-MM-dd') }}T00:00:00Z"
+                    top: 50
+                """
+        )
+    }
+)
+public class List extends Task implements RunnableTask<List.Output> {
+
+    @Schema(
+        title = "Authentication configuration",
+        description = "Microsoft Graph authentication settings including tenant ID, client ID, and client secret"
+    )
+    @Valid
+    @NotNull
+    private GraphAuthConfig auth;
+
+    @Schema(
+        title = "Folder ID",
+        description = "Email folder to retrieve messages from (inbox, sentitems, drafts, deleteditems, or folder ID)",
+        defaultValue = "inbox"
+    )
+    private Property<String> folderId;
+
+    @Schema(
+        title = "Filter",
+        description = "OData filter expression to filter the results (e.g., 'isRead eq false', 'from/emailAddress/address eq 'sender@example.com'')"
+    )
+    private Property<String> filter;
+
+    @Schema(
+        title = "Top",
+        description = "Maximum number of messages to return",
+        defaultValue = "50"
+    )
+    private Property<Integer> top;
+
+    @Schema(
+        title = "User email",
+        description = "Email address of the user whose mailbox to access (optional, uses authenticated user if not specified)"
+    )
+    private Property<String> userEmail;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        // Create Graph service client
+        GraphServiceClient graphClient = GraphService.createClientCredentialClient(auth, runContext);
+
+        // Render properties
+        String rFolder = runContext.render(folderId).as(String.class).orElse("inbox");
+        String rFilterExpression = filter != null ? runContext.render(filter).as(String.class).orElse(null) : null;
+        Integer rMaxResults = runContext.render(top).as(Integer.class).orElse(50);
+        String rUser = userEmail != null ? runContext.render(userEmail).as(String.class).orElse(null) : null;
+
+        // Use userPrincipalName from auth if userEmail not provided
+        if (rUser == null && auth.getUserPrincipalName() != null) {
+            rUser = runContext.render(auth.getUserPrincipalName()).as(String.class).orElse(null);
+        }
+
+        logger.info("Listing messages from rFolder '{}' for rUser: {} (max: {})", rFolder, rUser != null ? rUser : "current rUser", rMaxResults);
+
+        // Execute request using proper request configuration
+        MessageCollectionResponse messagesResponse;
+        if (rUser != null) {
```

Comment:
We you have very similar blocks like this in if and else please refactor it into a proper method for readability 🙏